### DISCUSSION TITLE
.h file not found

### DISCUSSION THREAD

**Discussion by JeppHu** [strongswan/strongswan#1746] [Q&A] [answered]
Android Studio Build Log

~/Documents/project/strongswan/src/frontends/android/app/src/main/jni/../../../../../../..//src/libstrongswan/plugins/openssl/openssl_ed_public_key.c:17:10: fatal error: 'openssl/evp.h' file not found
[arm64-v8a] Compile        : strongswan <= openssl_ed_private_key.c
#include <openssl/evp.h>
         ^~~~~~~~~~~~~~~
1 error generated.

I am sure all C files are in ~/Documents/project/strongswan/src/frontends/android/app/src/main/jni/openssl/openssl/
And  Android.mk file in ~/Documents/project/strongswan/src/frontends/android/app/src/main/jni/openssl/

LOCAL_PATH := $(call my-dir)
include $(CLEAR_VARS)
LOCAL_MODULE := libcrypto_static
LOCAL_SRC_FILES := $(TARGET_ARCH_ABI)/libcrypto.a
LOCAL_EXPORT_C_INCLUDES := $(LOCAL_PATH)/include
include $(PREBUILT_STATIC_LIBRARY)

How to link the C files to Project
<img width="447" alt="2023-06-21 18 02 45" src="https://github.com/strongswan/strongswan/assets/25788362/ed41a9a1-3ca3-4d62-b2e3-82a3defc6cd5">

**Comment by tobiasbrunner** [comment, marked as answer]
> I am sure all C files are in ~/Documents/project/strongswan/src/frontends/android/app/src/main/jni/openssl/**openssl**

Which clearly isn't the correct directory:

> LOCAL_EXPORT_C_INCLUDES := $(LOCAL_PATH)/**include**
